gpcontrib/arenadata_toolkit/arenadata_toolkit--1.3--1.4.sql

```diff
@@ -0,0 +1,375 @@
+		                       n.nspname = 'arenadata_toolkit'
+		LEFT JOIN pg_attribute a ON a.attrelid = c.oid AND
+		                            a.attname = 'tablespace_location'
+		WHERE c.relname = '__db_files_current')
+	THEN
+		CREATE OR REPLACE VIEW arenadata_toolkit.__db_files_current
+		AS
+		SELECT
+			c.oid AS oid,
+			c.relname AS table_name,
+			n.nspname AS table_schema,
+			c.relkind AS type,
+			c.relstorage AS storage,
+			d.datname AS table_database,
+			t.spcname AS table_tablespace,
+			dbf.segindex AS content,
+			dbf.segment_preferred_role AS segment_preferred_role,
+			dbf.hostname AS hostname,
+			dbf.address AS address,
+			dbf.full_path AS file,
+			dbf.size AS file_size,
+			dbf.modified_dttm AS modifiedtime,
+			dbf.changed_dttm AS changedtime,
+			CASE
+				WHEN 'pg_default' = t.spcname THEN gpconf.datadir || '/base'
+				WHEN 'pg_global' = t.spcname THEN gpconf.datadir || '/global'
+				ELSE (SELECT tblspc_loc
+					  FROM gp_tablespace_segment_location(t.oid)
+					  WHERE gp_segment_id = dbf.segindex)
+				END AS tablespace_location
+		FROM arenadata_toolkit.__db_segment_files dbf
+		LEFT JOIN pg_class c ON c.oid = dbf.reloid
+		LEFT JOIN pg_namespace n ON c.relnamespace = n.oid
+		LEFT JOIN pg_tablespace t ON dbf.tablespace_oid = t.oid
+		LEFT JOIN pg_database d ON dbf.datoid = d.oid
+		LEFT JOIN gp_segment_configuration gpconf ON dbf.dbid = gpconf.dbid;
+	END IF;
+END $$;
+
+DO $$
+BEGIN
+	IF (SELECT a.attrelid IS NULL
+		FROM pg_class c
+		JOIN pg_namespace n ON n.oid = c.relnamespace AND
+		                       n.nspname = 'arenadata_toolkit'
+		LEFT JOIN pg_attribute a ON a.attrelid = c.oid AND
+		                            a.attname = 'tablespace_location'
+		WHERE c.relname = '__db_files_current_unmapped')
+	THEN
+		CREATE OR REPLACE VIEW arenadata_toolkit.__db_files_current_unmapped
+		AS
+		SELECT
+			v.table_database,
+			v.table_tablespace,
+			v.content,
+			v.segment_preferred_role,
+			v.hostname,
+			v.address,
+			v.file,
+			v.file_size,
+			v.tablespace_location
+		FROM arenadata_toolkit.__db_files_current v
+		WHERE v.oid IS NULL;
+	END IF;
+END $$;
+
+DO $$
+BEGIN
+	IF (SELECT a.attrelid IS NULL
+		FROM pg_class c
+		JOIN pg_namespace n ON n.oid = c.relnamespace AND
+		                       n.nspname = 'arenadata_toolkit'
+		LEFT JOIN pg_attribute a ON a.attrelid = c.oid AND
+		                            a.attname = 'tablespace_location'
+		WHERE c.relname = 'db_files_history')
+	THEN
+		EXECUTE FORMAT($fmt$ALTER TABLE arenadata_toolkit.db_files_history
+							RENAME TO %1$I;$fmt$,
+			to_char(now(), '"db_files_history_backup_"YYYYMMDD"t"HH24MISS'));
+	END IF;
+END $$;
+
+/*
+ This is part of arenadata_toolkit API for ADB Bundle.
+ This function collects information in db_files_current and db_files_history tables.
+ */
+CREATE OR REPLACE FUNCTION arenadata_toolkit.adb_collect_table_stats()
+RETURNS VOID
+AS $$
+BEGIN
+	IF NOT EXISTS (
+		SELECT 1
```

Comment:
when the exists clause is used, then select's target list does not matter. So remove please '1', use empty target list.